NadeoServices.as

```diff
@@ -155,9 +156,14 @@ namespace NadeoServices
 			}
 		}
 
-		if (missing.Length > 0) {
+		if (missing.Length == 0) {
+			return ret;
+		}
+
+		while (true) {
```

Comment:
Why not do this instead:
```asc
while (missing.Length > 0) {
```
Should be more concise, you can simplify the exit condition at the end of the loop, and you can get rid of the `Length == 0` check above!

Reply:
You're right, I've simplified that now